pkg/tsdb/sqleng/sql_engine_test.go

```diff
@@ -9,317 +9,308 @@ import (
+		fixtures[6] = nilPointer
+
+		for i := range fixtures {
+			ConvertSqlTimeColumnToEpochMs(fixtures, i)
+		}
+
+		t.Run("When converting them should return epoch time with millisecond precision", func(t *testing.T) {
+			require.Equal(t, tSeconds*1e3, fixtures[0].(int64))
+			require.Equal(t, tSeconds*1e3, fixtures[1].(int64))
+			require.Equal(t, tMilliseconds, fixtures[2].(int64))
+			require.Equal(t, tMilliseconds, fixtures[3].(int64))
+			require.Equal(t, tMilliseconds, fixtures[4].(int64))
+			require.Equal(t, tMilliseconds, fixtures[5].(int64))
+			require.Nil(t, fixtures[6])
 		})
+	})
 
-		Convey("Given row values with time.Time as time columns", func() {
-			var nilPointer *time.Time
+	t.Run("Given row values with uint64 as time columns", func(t *testing.T) {
+		tSeconds := uint64(dt.Unix())
+		tMilliseconds := uint64(dt.UnixNano() / 1e6)
+		tNanoSeconds := uint64(dt.UnixNano())
+		var nilPointer *uint64
+
+		fixtures := make([]interface{}, 7)
+		fixtures[0] = tSeconds
+		fixtures[1] = &tSeconds
+		fixtures[2] = tMilliseconds
+		fixtures[3] = &tMilliseconds
+		fixtures[4] = tNanoSeconds
+		fixtures[5] = &tNanoSeconds
+		fixtures[6] = nilPointer
+
+		for i := range fixtures {
+			ConvertSqlTimeColumnToEpochMs(fixtures, i)
+		}
+
+		t.Run("When converting them should return epoch time with millisecond precision", func(t *testing.T) {
+			require.Equal(t, int64(tSeconds*1e3), fixtures[0].(int64))
+			require.Equal(t, int64(tSeconds*1e3), fixtures[1].(int64))
+			require.Equal(t, int64(tMilliseconds), fixtures[2].(int64))
+			require.Equal(t, int64(tMilliseconds), fixtures[3].(int64))
+			require.Equal(t, int64(tMilliseconds), fixtures[4].(int64))
+			require.Equal(t, int64(tMilliseconds), fixtures[5].(int64))
+			require.Nil(t, fixtures[6])
+		})
+	})
 
-			fixtures := make([]interface{}, 5)
-			fixtures[0] = dt
-			fixtures[1] = &dt
-			fixtures[2] = earlyDt
-			fixtures[3] = &earlyDt
-			fixtures[4] = nilPointer
+	t.Run("Given row values with int32 as time columns", func(t *testing.T) {
+		tSeconds := int32(dt.Unix())
+		var nilInt *int32
 
-			for i := range fixtures {
-				ConvertSqlTimeColumnToEpochMs(fixtures, i)
-			}
+		fixtures := make([]interface{}, 3)
+		fixtures[0] = tSeconds
+		fixtures[1] = &tSeconds
+		fixtures[2] = nilInt
 
-			Convey("When converting them should return epoch time with millisecond precision ", func() {
-				expected := float64(dt.UnixNano()) / float64(time.Millisecond)
-				expectedEarly := float64(earlyDt.UnixNano()) / float64(time.Millisecond)
+		for i := range fixtures {
+			ConvertSqlTimeColumnToEpochMs(fixtures, i)
+		}
 
-				So(fixtures[0].(float64), ShouldEqual, expected)
-				So(fixtures[1].(float64), ShouldEqual, expected)
-				So(fixtures[2].(float64), ShouldEqual, expectedEarly)
-				So(fixtures[3].(float64), ShouldEqual, expectedEarly)
-				So(fixtures[4], ShouldBeNil)
-			})
+		t.Run("When converting them should return epoch time with millisecond precision", func(t *testing.T) {
```

Comment:
Do you mind just removing t.Run here? Every t.Run introduces a sub-test, and I don't think it makes sense to group assertions as sub-tests. IMO a sub-test should correspond to a test case.